src/NzbDrone.Core/DecisionEngine/Specifications/UpgradableSpecification.cs

```diff
@@ -57,9 +90,48 @@ private bool IsPreferredWordUpgradable(int currentScore, int newScore)
             return newScore > currentScore;
         }
 
+        public bool IsUpgradable(QualityProfile qualityProfile, LanguageProfile languageProfile, List<QualityModel> currentQualities, List<Language> currentLanguages, int currentScore, QualityModel newQuality, Language newLanguage, int newScore)
+        {
+            if (IsQualityUpgradable(qualityProfile, currentQualities, newQuality))
+            {
+                return true;
+            }
+
+            foreach (var quality in currentQualities)
```

Comment:
Rather than looping over the qualities again, would it be easier to follow if `IsQualityUpgradable` return `-1`, `0` or `1`  depending on whether it's a downgrade, no change or upgrade?  Then we return true on upgrade (`1`), false on downgrade `-1`, and let it fall through to language on no change (`0`)?

It feels like we're almost doing the same calculation twice and there's scope for the two to get out of sync.

I might have missed the point again though...